### DISCUSSION TITLE
2024 Cohort 2 - Group 15 Discussion

### DISCUSSION THREAD

**Discussion by sharon-digipres** [Digital-Preservation-Coalition/PythonStudyGroups#41] [General]
This is the discussion thread for those in **Group 15** of the July-December 2024 Cohort

**Comment by barsness13** [comment]
Hi Group 15!  Here's our Google Drive: https://drive.google.com/drive/folders/1qHFiI_FfrsTfRI2syG3TxXlFWqnqKrb-?usp=sharing

And here are our meeting notes: https://docs.google.com/document/d/14QzVrJAHRsNPxUcGOvOSTDaNYMevRhfQ0Hp7QWhhUvw/edit

If you'd like to use a different email to access the Google content, just let me know.

**Comment by barsness13** [comment]
I've been spending some time working on my project and while I don't have a working script yet, I thought it might be fun/helpful for me to share the bit I've been using to get a starting directory from user, validate it, and recursively search that directory to get a list of all the files in that directory and its sub-directories:

```
#get starting directory from user, validate, create a recursive file list
dirinput=input('Starting directory: ')
if os.path.isdir(dirinput)==False:
    print()
    print('PROBLEM WITH STARTING DIRECTORY! It looks like',dirinput,'is not a valid location. Please make sure your file path is complete and does NOT end with a final slash.')
    quit()
else:
    filelist=[]
    def list_files_recursive(path='.'):
        for entry in os.listdir(path):
            full_path = os.path.join(path, entry)
            if os.path.isdir(full_path):
                list_files_recursive(full_path)
            else:
                filelist.append(full_path)
    list_files_recursive(dirinput)

```
from there, my script is going to iterate through each file in the list and pull different kinds of metadata from different libraries and put it all in a single dictionary.

**Comment by mpstrickland** [comment]
[heart]         Manuela Pallotto Strickland reacted to your message:
________________________________

**Comment by mpstrickland** [comment]
Hi all,
looking forward to seeing you later today (if I got the time/date correct...).
I have spent these last couple of weeks trying to get *.py scripts executable. I am not sure what the issue is, really. I edited the PATH variables and added one which points to the directory where I am keeping these testing scripts for this group but not much changed. I managed to run only a simple helloworld.py so the most basic/obvious things (is python installed? Is the PATH set up to execute?) seems to have positive answers. I honestly do not know if the issue is with the scripts (which I retrieved online) or my machine set up. There is no way for me to find out, as I cannot possibly assess/validate these codes, I would not know where to start from.
In the attempt to make my task simpler, I decided to focus on extracting EXIF from images, nothing else. I have put each script in the same folder as the image (which is also the folder where the helloworld.py sits), made it executable [chmod +x *.py], then run it [python3 *.py]. Nothing happens at all, with none of them, although allegedly they should output something, at least on screen. I do know for a fact that the TIFs have EXIF because I can see it with IM and GIMP. As a last attempt, I run these scripts in the interpreter line by line, to see where the code stops working, but to understand how to correct the errors that are thrown feels a bit overwhelming, and most of the time I have not been able to understand how I should fix the lines. I am basically positive that there is something that I should do to these scripts to make them work, but I don't know what it is. I am not sure I will give any contribution to our meeting, sorry!!! But I am looking forward to see the wonderful things that you all have been doing. :)

**Comment by barsness13** [comment]
All, I'm afraid something unexpected has come up last-minute and I won't be able to make our meeting. I'm looking forward to reviewing the notes and seeing how far you all have gotten! I'm not sure I got too much further than the snippit I shared a while ago, but I'll also share what code I have once I'm back online.  If you also have bits of code you'd like to share, you can always drop them in the Google Drive folder -- and if you have trouble accessing it just let me know!

**Comment by mpstrickland** [comment]
[sad]   Manuela Pallotto Strickland reacted to your message:
________________________________

**Comment by mpstrickland** [comment]
Hi all in Group 15, I am in the zoom waiting room, hopefully I will get in soon :)

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections
King's People: Manuela Pallotto Strickland | King's College London (kcl.ac.uk)<https://www.kcl.ac.uk/people/manuela-pallotto-strickland>

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

I may send emails outside of normal working hours at times that are convenient for me. Please do not feel any pressure to respond outside of your own normal working hours at times that are inconvenient for you.

**Comment by mpstrickland** [comment]
Hi Group 15! I am not managing to get into zoom, it keeps crashing. I am going to try to join it in the web browser. Fingers crossed!
Manuela

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections
King's People: Manuela Pallotto Strickland | King's College London (kcl.ac.uk)<https://www.kcl.ac.uk/people/manuela-pallotto-strickland>

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

I may send emails outside of normal working hours at times that are convenient for me. Please do not feel any pressure to respond outside of your own normal working hours at times that are inconvenient for you.

***@***.***

**Comment by barsness13** [comment]
A quick update for you all that I have a *mostly* working script loaded into our Google Drive: https://drive.google.com/file/d/1aAXwNNUVFL1bv4FiarTNfKBtPtLos3_F/view?usp=drive_link  There's something buggy happening with the xmp metadata section -- it will run just fine but it never pulls the metadata out, even when I know it should be there.  My goal for the next month is to get that section working.

**Comment by barsness13** [comment]
Also, the topic of Google Colab came up today -- it's basically a Google version of Jupyter Notebooks.  They're nice because you don't need to have Python on your computer to use it, and you can add text around your scripts to make them a bit easier for non-Python folks to use.  As an example, here's a script I wrote in GitHub: https://github.com/barsness13/digipres-python-scripts/blob/main/rssharvester.py and here's the same script re-written for Colab: https://colab.research.google.com/drive/14GR7YcNTpbJYmdCz1iFs3xopmuK_izKV?usp=sharing

**Comment by mpstrickland** [comment]
[heart]         Manuela Pallotto Strickland reacted to your message:
________________________________

**Comment by mpstrickland** [comment]
Hello Group 15! I am due to cover the archives' reading room in 30 minutes and for the rest of the afternoon because we are very much short of staff today. I will try to join in our meeting, albeit silently, but I might not be able to, it all depends on how busy it is. If you don't see me and I don't see you, happy new year!
